File: PHP/Endpoint Examples/JSON Payload/pdf-with-redacted-text-preview.php
<?php
require 'vendor/autoload.php'; // Require the autoload file to load Guzzle HTTP client.

use GuzzleHttp\Client; // Import the Guzzle HTTP client namespace.
use GuzzleHttp\Psr7\Request; // Import the PSR-7 Request class.
use GuzzleHttp\Psr7\Utils; // Import the PSR-7 Utils class for working with streams.

// By default, we use the US-based API service. This is the primary endpoint for global use.
$apiUrl = "https://api.pdfrest.com";

/* For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
 * For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 */
//$apiUrl = "https://eu-api.pdfrest.com";

// Toggle deletion of sensitive files (default: false)
$DELETE_SENSITIVE_FILES = false;

$upload_client = new Client(['http_errors' => false]);
$upload_headers = [
  'api-key' => 'xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  'content-filename' => 'filename.pdf',
  'Content-Type' => 'application/octet-stream'
];
$upload_body = file_get_contents('/path/to/file');
$upload_request = new Request('POST', $apiUrl.'/upload', $upload_headers, $upload_body);
$upload_res = $upload_client->sendAsync($upload_request)->wait();
echo $upload_res->getBody() . PHP_EOL;

$upload_response_json = json_decode($upload_res->getBody());

$uploaded_id = $upload_response_json->{'files'}[0]->{'id'};

echo "Successfully uploaded with an id of: " . $uploaded_id . PHP_EOL;

$redact_text_client = new Client(['http_errors' => false]);
$redaction_options = '[{\"type\":\"preset\",\"value\":\"email\"},{\"type\":\"regex\",\"value\":\"(\\\\\\\\+\\\\\\\\d{1,2}\\\\\\\\s)?\\\\\\\\(?\\\\\\\\d{3}\\\\\\\\)?[\\\\\\\\s.-]\\\\\\\\d{3}[\\\\\\\\s.-]\\\\\\\\d{4}\"},{\"type\":\"literal\",\"value\":\"word\"}]';
$redact_text_headers = [
  'api-key' => 'xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  'Content-Type' => 'application/json'
];
$redact_text_body = '{"id":"'.$uploaded_id.'", "redactions":"'.$redaction_options.'"}';
$redact_text_request = new Request('POST', $apiUrl.'/pdf-with-redacted-text-preview', $redact_text_headers, $redact_text_body);
$redact_text_res = $redact_text_client->sendAsync($redact_text_request)->wait();
$preview_body_str = (string)$redact_text_res->getBody();
echo $preview_body_str . PHP_EOL;

// All files uploaded or generated are automatically deleted based on the
// File Retention Period as shown on https://pdfrest.com/pricing.
// For immediate deletion of files, particularly when sensitive data
// is involved, an explicit delete call can be made to the API.
//
// Deletes all files in the workflow, including outputs. Save all desired files before enabling this step.
// IMPORTANT: Do not delete the $preview_id (the preview PDF) file until after the redaction is applied
// with the /pdf-with-redacted-text-applied endpoint.

if ($DELETE_SENSITIVE_FILES) {
  $delete_client = new Client(['http_errors' => false]);
  $delete_headers = [
    'api-key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
    'Content-Type' => 'application/json'
  ];
  $preview_json = json_decode($preview_body_str, true);
  $preview_id = isset($preview_json['outputId']) ? $preview_json['outputId'] : '';
  $delete_body = json_encode([ 'ids' => $uploaded_id . ', ' . $preview_id ]);
  $delete_request = new Request('POST', $apiUrl.'/delete', $delete_headers, $delete_body);
  $delete_res = $delete_client->sendAsync($delete_request)->wait();
  echo $delete_res->getBody() . PHP_EOL;
}

File: DotNET/Endpoint Examples/JSON Payload/pdf-with-redacted-text-preview.cs

/*
 * What this sample does:
 * - Called from Program.cs to preview text redactions on a PDF via JSON flow.
 *
 * Setup (environment):
 * - Copy .env.example to .env
 * - Set PDFREST_API_KEY=your_api_key_here
 * - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
 *     PDFREST_URL=https://eu-api.pdfrest.com
 *   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 *
 * Usage:
 *   dotnet run -- pdf-with-redacted-text-preview /path/to/input.pdf
 *
 * Output:
 * - Prints JSON responses; non-2xx results exit non-zero.
 */
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System.Text;

namespace Samples.EndpointExamples.JsonPayload
{
    public static class PdfWithRedactedTextPreview
    {
        public static async Task Execute(string[] args)
        {
            if (args == null || args.Length < 1) { Console.Error.WriteLine("pdf-with-redacted-text-preview requires <inputFile>"); Environment.Exit(1); return; }
            var inputPath = args[0]; if (!File.Exists(inputPath)) { Console.Error.WriteLine($"File not found: {inputPath}"); Environment.Exit(1); return; }
            var apiKey = Environment.GetEnvironmentVariable("PDFREST_API_KEY"); if (string.IsNullOrWhiteSpace(apiKey)) { Console.Error.WriteLine("Missing required environment variable: PDFREST_API_KEY"); Environment.Exit(1); return; }
            var baseUrl = Environment.GetEnvironmentVariable("PDFREST_URL") ?? "https://api.pdfrest.com";
            using (var httpClient = new HttpClient { BaseAddress = new Uri(baseUrl) })
            using (var uploadRequest = new HttpRequestMessage(HttpMethod.Post, "upload"))
            {
                uploadRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                uploadRequest.Headers.Accept.Add(new("application/json"));
                var uploadByteArray = File.ReadAllBytes(inputPath);
                var uploadByteAryContent = new ByteArrayContent(uploadByteArray);
                uploadByteAryContent.Headers.TryAddWithoutValidation("Content-Type", "application/octet-stream");
                uploadByteAryContent.Headers.TryAddWithoutValidation("Content-Filename", Path.GetFileName(inputPath));
                uploadRequest.Content = uploadByteAryContent;
                var uploadResponse = await httpClient.SendAsync(uploadRequest);
                var uploadResult = await uploadResponse.Content.ReadAsStringAsync();
                Console.WriteLine("Upload response received.");
                Console.WriteLine(uploadResult);
                JObject uploadResultJson = JObject.Parse(uploadResult);
                var uploadedID = uploadResultJson["files"][0]["id"];
                using (var redactedTextRequest = new HttpRequestMessage(HttpMethod.Post, "pdf-with-redacted-text-preview"))
                {
                    redactedTextRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                    redactedTextRequest.Headers.Accept.Add(new("application/json"));
                    redactedTextRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");
                    var redaction_option_array = new JArray
                    {
                        new JObject { ["type"] = "regex", ["value"] = "(?:\\(\\d{3}\\)\\s?|\\d{3}[-.\\s]?)?\\d{3}[-.\\s]?\\d{4}" },
                        new JObject { ["type"] = "literal", ["value"] = "word" }
                    };
                    JObject parameterJson = new JObject { ["id"] = uploadedID, ["redactions"] = JsonConvert.SerializeObject(redaction_option_array) };
                    redactedTextRequest.Content = new StringContent(parameterJson.ToString(), Encoding.UTF8, "application/json");
                    var redactedTextResponse = await httpClient.SendAsync(redactedTextRequest);
                    var redactedTextResult = await redactedTextResponse.Content.ReadAsStringAsync();
                    Console.WriteLine("Processing response received.");
                    Console.WriteLine(redactedTextResult);

                    // All files uploaded or generated are automatically deleted based on the 
                    // File Retention Period as shown on https://pdfrest.com/pricing. 
                    // For immediate deletion of files, particularly when sensitive data 
                    // is involved, an explicit delete call can be made to the API.
                    //
                    // Deletes all files in the workflow, including outputs. Save all desired files before enabling this step.
                    // IMPORTANT: Do not delete the previewId (the preview PDF) file until after the redaction is applied
                    // with the /pdf-with-redacted-text-applied endpoint.

                    // Optional immediate deletion of sensitive files
                    // Default: false; override with PDFREST_DELETE_SENSITIVE_FILES=true
                    var deleteSensitiveFiles = string.Equals(
                        Environment.GetEnvironmentVariable("PDFREST_DELETE_SENSITIVE_FILES"),
                        "true",
                        StringComparison.OrdinalIgnoreCase);

                    if (deleteSensitiveFiles)
                    {
                        using (var deleteRequest = new HttpRequestMessage(HttpMethod.Post, "delete"))
                        {
                            deleteRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                            deleteRequest.Headers.Accept.Add(new("application/json"));
                            deleteRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");

                            var previewId = JObject.Parse(redactedTextResult)["outputId"].ToString();
                            JObject deleteJson = new JObject { ["ids"] = uploadedID + ", " + previewId };
                            deleteRequest.Content = new StringContent(deleteJson.ToString(), Encoding.UTF8, "application/json");
                            var deleteResponse = await httpClient.SendAsync(deleteRequest);
                            var deleteResult = await deleteResponse.Content.ReadAsStringAsync();
                            Console.WriteLine(deleteResult);
                        }
                    }
                }
            }
        }
    }
}
